src/Hosting/Hosting/src/WebHostExtensions.cs

```diff
@@ -159,14 +159,8 @@ private static async Task WaitForTokenShutdownAsync(this IWebHost host, Cancella
         },
         applicationLifetime);
 
-        var waitForStop = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
-        applicationLifetime.ApplicationStopping.Register(obj =>
-        {
-            var tcs = (TaskCompletionSource)obj!;
-            tcs.TrySetResult();
-        }, waitForStop);
-
-        await waitForStop.Task;
+        await Task.Delay(Timeout.Infinite, applicationLifetime.ApplicationStopping)
+            .ConfigureAwait(ConfigureAwaitOptions.SuppressThrowing | ConfigureAwaitOptions.ContinueOnCapturedContext);
```

Comment:
No, `ForceYielding` is for the initial await but `RunContinuationsAsynchronously` is for when the task actually completes. But there's no need for a `RunContinuationsAsynchronously` analogue here because `Task.Delay` always runs cancellations asynchronously.